### ISSUE TITLE
no-redeclare broken when node environment is set

### ISSUE THREAD

**Issue by pigulla** [eslint/eslint#2064]
Using ESLint v0.17.0:

```js
// .eslintrc
{
    "env": {
        "node": false
    },

    "rules": {
        "no-unused-vars": 0
    }
}
```

```js
// test.js
var x = 42,
    x = 17;
```

```
$ node_modules/eslint/bin/eslint.js test.js

test.js
  2:4  error  x is already defined  no-redeclare

✖ 1 problem (1 error, 0 warnings)
```

Now when the `node` setting in `env` is changed to `true`, no error is reported. Happens in the online demo as well.

**Comment by nzakas** [comment]
Thanks, will take a look.

**Comment by gyandeeps** [comment]
I just pasted this on online demo
```.js
/*eslint-env node */
var x = 42,
    x = 17;
```

It showed the error.

**Comment by btmills** [comment]
Interestingly, I was only able to repro by setting the environment in a config file, not inline.

**Comment by nzakas** [comment]
@btmills that might be a bug - they should work the same.

**Comment by nzakas** [comment]
Working on this.
